auth_server/main.go

```diff
@@ -129,22 +128,25 @@ func ServeOnce(c *server.Config, cf string, hd *httpdown.HTTP) (*server.AuthServ
 		glog.Warning("Running without TLS")
 		tlsConfig = nil
 	}
+
 	hs := &http.Server{
 		Addr:      c.Server.ListenAddress,
 		Handler:   as,
 		TLSConfig: tlsConfig,
 	}
-
-	s, err := hd.ListenAndServe(hs)
-	if err != nil {
-		glog.Exitf("Failed to set up listener: %s", err)
-	}
+	go func() {
+		if err := hs.ListenAndServeTLS(c.Server.CertFile, c.Server.KeyFile); err != nil {
```

Comment:
I'm concerned this is a breaking change as it requires having a TLS cert/key (even if self-signed), as some users serve docker auth as plaintext and have a TLS reverse proxy in front of the service handling secure communication. Would you be able to add a conditional to run with and without certs?

Thanks for you PRs :)

Reply:
Thanks for the review. 
I didn't realise that my own configuration was cert/key with token and HTTP for server/reverse proxy.

Reply:
No worries, thanks for the quick turnaround :)

I'll checkout the other PR now.